torch/_ops.py

```diff
@@ -588,9 +698,45 @@ def handler(*args, **kwargs):
             add_cached_op(self)
             return handler
 
-        cache_result = True
         functionality_key = torch._C._to_functionality_key(key)  # type: ignore[attr-defined]
-        if functionality_key in mode_stack_per_key():
+        if functionality_key == torch._C.DispatchKey.PreDispatch:
+            curr_stack_len = _len_torch_dispatch_stack_pre_dispatch()
+            # The check for Python in the exclude set is so we properly respect `with no_dispatch()`
+            # calls inside of a mode.
+            if (
+                curr_stack_len > 0
+                and not torch._C._dispatch_tls_is_dispatch_key_excluded(
+                    DispatchKey.PreDispatch
+                )
+            ):
+
+                def handler(*args, **kwargs):
+                    @contextlib.contextmanager
```

Comment:
Nit: using `contextlib.contextmanager`, but `_temporarily_pop_modes_from_pre_dispatch` isn't actually being used as a context manager

Reply:
Hmm i think we are immediately after its' definition?

Reply:
oh nevermind. I was thinking that it had to be used as a decorator